mautic/mautic
# NPM dependencies part 2

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [enhancement]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR is continuing what https://github.com/mautic/mautic/pull/12234 started. That PR must be merged first.

The big refactoring here concerns the Bootstrap library which is the CSS framework that the whole administration is built on.

I'm doing my best to install the same JS library versions as we had before but in cases it's not possible:
- Minicolors were updated to the oldest version published on NPM. There was a bug where the color hex value was outputing with the leading `#` and Mautic expected values without it. So I added an "on blur" listener to remove the hashes from the color fields.
- Multiselect was updated from version 0.9.11 to 0.9.12 as that's the only [version installable via NPM](https://www.npmjs.com/package/multiselect?activeTab=versions)

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester)). If you are testing on your local environment run `composer install` to download the dependencies before testing. Gitpod is doing that automatically.
2. Browse through the UI. Nothing should change. Colors, dimensions, icons should still be the same.
3. Check that modal boxes work.
4. Check color select boxes in the category edit form and focus items.
5. Check the date-time pickers that they work the same as before.
6. Asset upload
7. Test multiselect for example when creating a segment with the "segment membership" filter.
8. Charts on the email or page detail pages are showing up as before
9. Test the email and page builder
10. Test that for example segment edit page has CK editor loaded for the description field
11. Test also the campaign builder
12. There are no errors in the browser's JS console.

I also tested that the Gated Video feature still works:
![Screenshot 2023-04-26 at 12 36 54](https://user-images.githubusercontent.com/1235442/234550578-8a3bd74b-430e-4928-84f2-a806a263069a.png)
It's a bit harder to set up: https://docs.mautic.org/en/gated-video
